lib/oidc/pollTokenWithCiba.ts

```diff
@@ -0,0 +1,48 @@
+/*!
+ * Copyright (c) 2015-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * 
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/* eslint-disable camelcase */
+import { getOAuthUrls } from './util/oauth';
+import { AuthSdkError } from './../errors';
+import { 
+  OktaAuthOAuthInterface, 
+  OAuthResponse, 
+  CibaTokenOptions, 
+  TokenParamsProto,
+  ClientAuthenticationOptions,
+} from './types';
+import { postToTokenEndpoint } from './endpoints';
+import { prepareClientAuthenticationParams } from './util/prepareClientAuthenticationParams';
+
+const GRANT_TYPE = 'urn:openid:params:grant-type:ciba';
+
+export async function pollTokenWithCiba(
```

Comment:
I think we should name this function something like `confirmCibaAuthReq` and document that this method is designed to be used with a polling pattern. The method itself does not "poll", so I think using the word `poll` in the name may lead to confusion

Reply:
I wonder if we want this method to throw when then auth req is still pending. We may want to catch the "pending" error and return a `status` of some kind back (for success and pending). Especially since this method is designed to be used in a polling pattern